scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -305,8 +305,13 @@ class ScioContext private[scio] (val options: PipelineOptions,
     if (_pipeline != null) {
       throw new RuntimeException("Cannot set job name once pipeline is initialized")
     }
-    Try(optionsAs[DataflowPipelineOptions])
-      .foreach(_.setJobName(new DataflowPipelineOptions.JobNameFactory().create(options)))
+    Try(optionsAs[DataflowPipelineOptions]).foreach { opts =>
+      if (name != "") {
```

Comment:
How about just passing the name, and delegating the checks to the sdk (which happens locally, before the job is submitted)?